01-data-structures/01-introduction-to-data-structures/screen/screen.rb

```diff
@@ -6,18 +6,23 @@ class Screen
   attr_accessor :matrix
 
   def initialize(width, height)
+    self.width = width
+    self.height = height
+    self.matrix = Hash.new
   end
 
   # Insert a Pixel at x, y
   def insert(pixel, x, y)
+    self.matrix.merge!([x, y] => pixel )
```

Comment:
This works, but it is not optimal; you can simply do `@matrix[[x, y]] = pixel`. Your method creates a new `Hash` to merge with your current `Hash` every time, which is unnecessary.